src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/InfrastructureFormula.kt

```diff
@@ -59,16 +72,17 @@ class InfrastructureFormula<out T : VirtualUsers>(
         }
 
         val network = networkProvisioning.get()
-        val provisionJira = executor.submitWithLogContext("jira") {
-            overrideJiraNetwork(network).provision(
+
+        val provisionedJira =  userProvisionedJira ?: executor.submitWithLogContext("jira") {
+            overrideJiraNetwork(network)!!.provision(
```

Comment:
Now it started throwing NPEs

Reply:
public constructors prevents NPE here

Reply:
Indeed the constructors plus the elvis null-check might indeed prevent it. It's a few moving parts working together to prevent the NPE.